src/data/polynomial/eval.lean

```diff
@@ -306,9 +306,10 @@ eval₂_monomial _ _
 
 @[simp] lemma eval_bit1 : (bit1 p).eval x = bit1 (p.eval x) := eval₂_bit1 _ _
 
-@[simp] lemma eval_smul (p : R[X]) (x : R) {s : R} :
-  (s • p).eval x = s * p.eval x :=
-eval₂_smul (ring_hom.id _) _ _
+@[simp] lemma eval_smul [monoid S] [distrib_mul_action S R] [is_scalar_tower S R R]
+  (p : R[X]) (x : R) {s : S} :
```

Comment:
```suggestion
  (s : S) (p : R[X]) (x : R) :
```
To match the order they appear in the statement

Reply:
Does `smul_comp` also need this change?